### DISCUSSION TITLE
How to implement time dependent velocity field

### DISCUSSION THREAD

**Discussion by brandon-barclay** [idaholab/moose#21542] [Q&A General] [answered]
Hello, I am trying to implement a time dependent velocity field for an outdoor airflow simulation and I was looking for some advice regarding the best way to go about this. I found .csv data which contains wind velocity at given x,y,z locations and t. 

The only object I was able to find which appears to allow interpolation of data like this was the " PiecewiseMulticonstant " object, however it was not clear to me how to implement different function values at the different times. If anyone has any advice regarding this object, or a better way to interpolate the velocity data, it would be greatly appreciated.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Is the time dependence fully separable from the space dependence? (eg is it v(x,t) = f(t) g(x) ) ?
This would make it easy.

If not:
Is there an analytical equation for the data?
then you could use a parsed expression instead of a tabulation

and if not:
PiecewiseMultiConstant and PiecewiseMultiLinear are what you want to use. It is tabulated along X Y Z and T axis. There are example files in:
https://github.com/idaholab/moose/tree/next/test/tests/functions/piecewise_multilinear

Guillaume
